node/src/components/consensus/protocols/highway/tests.rs

```diff
@@ -233,3 +258,113 @@ fn max_rounds_per_era_returns_the_correct_value_for_prod_chainspec_value() {
 
     assert_eq!(219, max_rounds_per_era);
 }
+
+#[test]
+fn no_slow_down_when_all_nodes_fast() {
+    let mut validators = BTreeMap::new();
+    validators.insert(
+        ALICE_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*ALICE_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        BOB_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*BOB_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        CAROL_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*CAROL_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        DAVE_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*DAVE_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        ELLEN_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*ELLEN_SECRET_KEY)), 100),
+    );
+
+    let mut env = ConsensusEnvironment::new(validators, Default::default());
+    for _ in 0..10 {
+        env.crank_round();
+    }
+
+    assert_eq!(env.our_round_exp(), 0);
+}
+
+#[ignore = "TODO: unignore when exponent switching is improved"]
+#[test]
+fn slow_node_should_switch_own_round_exponent() {
+    let mut validators = BTreeMap::new();
+    validators.insert(
+        ALICE_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*ALICE_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        BOB_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*BOB_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        CAROL_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*CAROL_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        DAVE_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*DAVE_SECRET_KEY)), 100),
+    );
+    validators.insert(
+        ELLEN_PUBLIC_KEY.clone(),
+        (Keypair::from(Arc::clone(&*ELLEN_SECRET_KEY)), 100),
+    );
+
+    // Alice is the tested node; it will be slow, and so it should switch to a higher round
+    // exponent
+    let mut slow_nodes = BTreeSet::new();
+    slow_nodes.insert(ALICE_PUBLIC_KEY.clone());
+
+    let mut env = ConsensusEnvironment::new(validators, slow_nodes);
+    for _ in 0..10 {
+        env.crank_round();
+    }
+
+    assert!(env.our_round_exp() > 0);
```

Comment:
We should probably have a test that sets up a scenario that's simple enough that the exponent behavior can be calculated by hand

Reply:
I think that would work better as a unit test for the `RoundSuccessMeter` (renamed to `PerformanceMeter` in the next PR, which isn't up yet). These tests here are supposed to be high level test only asserting a very general behavior, they are deliberately oblivious to the implementation details of how slowing down and speeding up work.